File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.c
bool searchMatrix(int** matrix, int matrixSize, int* matrixColSize, int target){
  
    int r = matrixSize - 1;
    int c = 0;

    while (r >= 0 && c < *matrixColSize) {

        if (matrix[r][c] == target) {
            return true;
        }
        else if (matrix[r][c] > target) {
            r--;
        }
        else {
            c++;
        }
    }
    return false;
}

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if not matrix or not matrix[0]:
            return False
        
        r = len(matrix) - 1
        c = 0
        
        while r >= 0 and c < len(matrix[0]):
            if matrix[r][c] == target:
                return True
            elif matrix[r][c] > target:
                r -= 1
            else:
                c += 1
                
        return False